### ISSUE TITLE
Next Dev Server Recompiling Indefinitely

### ISSUE THREAD

**Issue by jamesandersonwalsh** [contentlayerdev/contentlayer#501]
## Problem
Going through the installation & setup guide with the latest versions of both `next@13.4.7`, `contentlayer@0.3.4`, and `next-contentlayer@0.3.4`  is causing my dev server to re-compile indefinitely, which keeps the application from ever loading in the browser.

## Dependencies
```
"dependencies": {
    "@heroicons/react": "^2.0.18",
    "@types/node": "20.3.2",
    "@types/react": "18.2.14",
    "@types/react-dom": "18.2.6",
    "contentlayer": "0.3.4",
    "date-fns": "^2.30.0",
    "eslint": "8.43.0",
    "eslint-config-next": "13.4.7",
    "next": "13.4.7",
    "next-contentlayer": "0.3.4",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "typescript": "5.1.6"
  },
  "devDependencies": {
    "@pandacss/dev": "^0.5.0",
    "prettier": "^2.8.8"
  }
```

## To Reproduce
When I remove `withContentLayer` from my next.config.js the problem goes away, and the next dev server only recompiles exactly when it needs to. The page loads in the browser just fine. 
```
// const { withContentlayer } = require("next-contentlayer");

/** @type {import('next').NextConfig} */
const nextConfig = {
  swcMinify: true
}

// module.exports = withContentlayer(nextConfig);
module.exports = nextConfig;
```

When I wrap my default export in `withContentLayer` the re-compile behavior starts again, and none of my pages will load:

```
const { withContentlayer } = require("next-contentlayer");

/** @type {import('next').NextConfig} */
const nextConfig = {
  swcMinify: true
}

module.exports = withContentlayer(nextConfig);
```

**Comment by haydenbleasel** [comment]
I run a Next.js 13 App Dir + Contentlayer stack on at least 4 different projects with exactly the same config, but only one of them has this problem. Still looking for a reason, will chime in again if I find it.

**Comment by schickling** [comment]
Thanks for opening this issue. Can you please provide **minimal** a reproduction repo with instructions how to reproduce the problem you're experiencing? Thank you!

**Comment by jamesandersonwalsh** [comment]
Hi @schickling I finally had a free hour to take a look at this. So it turns out in trying to reproduce this issue I discovered that it wasnt contentlayer that was forcing NextJS to recompile, it was my `PandaCSS` setup. Panda was attempting to compile the markdown files in my `posts` directory that I setup as I was going through the Getting Started Guide. There is no meaningful work I can establish that needs to be done here, I was able to resolve the issue by improving the glob patterns PandaCSS is using to determine which files need to be hot reloaded.
